pkg/proc/native/threads_linux_riscv64.go

```diff
@@ -0,0 +1,324 @@
+package native
+
+import (
+	"bytes"
+	"debug/elf"
+	"fmt"
+	"os"
+	"syscall"
+	"unsafe"
+
+	sys "golang.org/x/sys/unix"
+
+	"github.com/go-delve/delve/pkg/proc"
+	"github.com/go-delve/delve/pkg/proc/linutil"
+	"golang.org/x/arch/riscv64/riscv64asm"
+)
+
+func (thread *nativeThread) fpRegisters() ([]proc.Register, []byte, error) {
+	var err error
+	var riscv64_fpregs linutil.RISCV64PtraceFpRegs
+
+	thread.dbp.execPtraceFunc(func() { err = ptraceGetFpRegset(thread.ID, &riscv64_fpregs) })
+	fpregs := riscv64_fpregs.Decode()
+
+	if err != nil {
+		err = fmt.Errorf("could not get floating point registers: %v", err.Error())
+	}
+
+	return fpregs, riscv64_fpregs.Fregs, err
+}
+
+func pokeMemory(pid int, addr uintptr, data []byte) (count int, err error) {
```

Comment:
Why is this needed instead of using the normal way we write to memory?

Reply:
Originally I think we might have a TOCTOU issue when emulate singlestep by insert a ebreak instruction, so I use it for sync the cache to avoid that, but seems that delve will stop every other threads while debugging, if so we don't need it.

Reply:
I just runs the test without this custom `pokeMemory`, and it's works well, so we can safely restore using `sys.PtracePokeData`.